server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -905,6 +906,68 @@ protected void toXContent(XContentBuilder builder, boolean includeDefaults) thro
                 });
         }
 
+        /**
+         * Defines a parameter that takes any of the values of an enumeration.
+         *
+         * @param name          the parameter name
+         * @param updateable    whether the parameter can be changed by a mapping update
+         * @param initializer   a function that reads the parameter value from an existing mapper
+         * @param defaultValue  the default value, to be used if the parameter is undefined in a mapping
+         * @param enumClass     the enumeration class the parameter takes values from
+         */
+        public static <T extends Enum<T>> Parameter<T> enumParam(
+            String name,
+            boolean updateable,
+            Function<FieldMapper, T> initializer,
+            T defaultValue,
+            Class<T> enumClass
+        ) {
+            Set<T> acceptedValues = EnumSet.noneOf(enumClass);
```

Comment:
`allOf`?

Reply:
Of course `allOf`. That was a silly left over from a copy-paste. Thanks